### DISCUSSION TITLE
Is there a way to check if a device is a Chromebook using ActionScript or ANE?

### DISCUSSION THREAD

**Discussion by koreaflash** [airsdk/Adobe-Runtime-Support#3361] [Q&A]
**Using the Actionscript Capabilities class, you can check the following information:**
avHardwareDisable: false
hasAccessibility: false
hasAudio: true
hasAudioEncoder: true
hasEmbeddedVideo: true
hasIME: true
hasMP3: true
hasPrinting: false
hasScreenBroadcast: false
hasScreenPlayback: false
hasStreamingAudio: true
hasStreamingVideo: true
hasTLS: true
hasVideoEncoder: true
isDebugger: false
isEmbeddedInAcrobat: false
language: ko
localFileReadDisable: false
manufacturer: Android Linux
maxLevelIDC: 5.1
os: Linux 5.10.216-24917-gf16c23f03eb6-grunt
pixelAspectRatio: 1
playerType: Desktop
screenColor: color
screenDPI: 160
screenResolutionX: 1366
screenResolutionY: 768
serverString: A=t&SA=t&SV=t&EV=t&MP3=t&AE=t&VE= ...
touchscreenType: finger
cpuArchitecture: x86_64
languages: ko-KR,en-US
supports32BitProcesses: true
supports64BitProcesses: true
version: AND 50,2,3,1

**You can get the following information via com.janumedia.ane.deviceinfo.DeviceInfoExtension ANE:**
brand: google
density: 1
densityDpi: 160
device: grunt_cheets
widthPixels: 412
heightPixels: 634
manufacturer: Google
model: grunt
sdkVersion: 30

With the above information, can we tell if the device is a Chromebook?
Is there any other way to get information that can confirm that it is a Chromebook...?

**Comment by marchbold** [comment]
I believe the correct way to do this is to check for the system feature: `org.chromium.arc.device_management` 

You can use the package manager extension to do this:

```as3
var isChromebook:Boolean = PackageManager.service.hasSystemFeature( "org.chromium.arc.device_management" );
```

Package Manager extension is available here: https://github.com/distriqt/ANE-PackageManager/ or via apm

```
apm install com.distriqt.PackageManager
```

**Reply by koreaflash** [reply]
Thank you very much.

**Reply by koreaflash** [reply]
The following error occurs during packaging:

Error occurred while packaging the application:


aapt tool failed:C:\Users\user\AppData\Local\Temp\a466e450-4722-44b6-8213-da1bdd61022b\savedstate-1.2.1-res\values\values.xml:3: error: Found tag id where item is expected

C:\Users\user\AppData\Local\Temp\a466e450-4722-44b6-8213-da1bdd61022b\lifecycle-viewmodel-2.6.1-res\values\values.xml:3: error: Found tag id where item is expected

C:\Users\user\AppData\Local\Temp\a466e450-4722-44b6-8213-da1bdd61022b\lifecycle-runtime-2.6.1-res\values\values.xml:3: error: Found tag id where item is expected

C:\Users\user\AppData\Local\Temp\a466e450-4722-44b6-8213-da1bdd61022b\activity-1.7.0-res\values\values.xml:4: error: Found tag id where item is expected

**Reply by marchbold** [reply]
Looks like an issue with the version of air you are using, I'd suggest updating to the latest and ensuring you have the latest android sdk installed.
